src/xtgeo/grid3d/_grid_import.py

```diff
@@ -49,26 +49,27 @@ def from_file(gfile, fformat=None, **kwargs):  # pylint: disable=too-many-branch
             "Invalid fformat: <{}>, options are {}".format(fformat, fflist)
         )
 
-    # work on file extension
-    _, fext = gfile.splitext(lower=True)
+    if not gfile.memstream:
+        # work on file extension
```

Comment:
Other classes, such as GridProperty and RegularSurface, uses `generic_format_by_proposal()`. I think all fformat guessing/passing should use the same method.

Reply:
Agree. I spotted that, but considered a separate issue/PR. However will fix here :+1: